<?xml version="1.0"?>
<launch>
  <arg name="device" default="R2000"/>

  <arg name="launch_pf_description" default="true"
       doc="Whether to include the pf_description bringup with this launch file." />
  <arg name="launch_rviz" default="true"
       doc="Whether to start RViz with this launch file." />

  <group if="$(arg launch_pf_description)">
    <include file="$(find pf_description)/launch/pf_bringup.launch" >
      <arg name="scanner" value="r2000" />
    </include>
  </group>

  <!-- R2000 Driver -->
  <node pkg="pf_driver" type="ros_main" name="r2000_node" output="screen">
    <param name="device" value="$(arg device)"/>
    <rosparam file="$(find pf_driver)/config/r2000_params.yaml" command="load"/>
  </node>

  <group if="$(arg launch_rviz)">
    <node type="rviz" name="rviz" pkg="rviz" args="-d $(find pf_driver)/rviz/r2000.rviz" />
  </group>
</launch>
